packages/loader/container-definitions/src/fluidPackage.ts

```diff
@@ -3,82 +3,72 @@
-    bin?: { [key: string]: string };
-    man?: string | string[];
-    repository?: string | { type: string; url: string };
-    scripts?: { [key: string]: string };
-    config?: { [key: string]: string };
-    dependencies?: { [key: string]: string };
-    devDependencies?: { [key: string]: string };
-    peerDependencies?: { [key: string]: string };
-    bundledDependencies?: { [key: string]: string };
-    optionalDependencies?: { [key: string]: string };
-    engines?: { node: string; npm: string };
-    os?: string[];
-    cpu?: string[];
-    private?: boolean;
+    /**
+     * The name of the target. For a browser environment, this could be umd for scripts
+     * or css for styles.
+     */
+    [target: string]: undefined | {
+        /**
+         * List of files. The file type in determined by library target
+         */
+        files: string[];
+
+        /**
+         * If libraryTarget is umd then library is the global name that the script entry points will be exposed
+         * under. Other target formats may choose to reinterpret this value.
+         */
+        library?: string;
+    }
 }
 
 /**
- * Fluid-specific properties expected on a package to be loaded by the Fluid code loader
+ * Fluid-specific properties expected on a package to be loaded by the  code loader.
+ * While compatible with the npm package format it is not necessary that that package is an
+ * npm package:
+ * https://stackoverflow.com/questions/10065564/add-custom-metadata-or-config-to-package-json-is-it-valid
  */
-export interface IFluidPackage extends IPackage {
-    // https://stackoverflow.com/questions/10065564/add-custom-metadata-or-config-to-package-json-is-it-valid
+export interface IFluidPackage {
+    /**
+     * The name of the package that this code represnets
+     */
+    name: string;
+    /**
+     * This object represents the Fluid specific properties of the package
+     */
     fluid: {
-        browser: {
-            [libraryTarget: string]: {
-                // List of bundled JS files. Absolute URLs will be loaded directly. Relative paths will be specific
-                // to the CDN location
-                files: string[];
-
-                // If libraryTarget is umd then library is the global name that the script entry points will be exposed
-                // under. Other target formats may choose to reinterpret this value.
-                library: string;
-            };
-        };
+        /**
+         * The name of the of the environment. This should be something like browser, or node
+         * and contain the necessary targets for loading this code in that environment.
+         */
+        [environment: string]:  undefined | IFluidPackageEnvironment;
     };
+    /**
+     * General access for extended fields as specific usages will
+     * likely have additional infornamation like a definition of
+     * compatible versions, or deployment information like rings or rollouts.
+     */
+    [key: string]: any;
```

Comment:
I don't really like this. It just gives intellisense for any property. I prefer the use-case of leaving this off and then having custom interfaces extend `IFluidPackage`, then they just cast as expected or feature-test for those properties.

Reply:
I had similar thoughts. for now, i'd like to keep it as a back compat crutch in case i missed a usage but would like to remove it in a follow. i wonder if i can mark it as deprecated